### PR TITLE
[PlSql] Support of create database

### PR DESCRIPTION
**Pull request by hellozrh**
[PlSql] Support of create database

### CODE REVIEW

**Comment by hellozrh** [comment]
[PlSql] Support of create database #3167

File: sql/plsql/PlSqlLexer.g4 [hunk 1]
```diff
@@ -788,6 +788,7 @@ LAYER:                        'LAYER';
 LDAP_REGISTRATION_ENABLED:    'LDAP_REGISTRATION_ENABLED';
 LDAP_REGISTRATION:            'LDAP_REGISTRATION';
 LDAP_REG_SYNC_INTERVAL:       'LDAP_REG_SYNC_INTERVAL';
+LEAF:                         'LEAF';
```
Comment (KvanTTT) [inline sql/plsql/PlSqlLexer.g4:791]: Please add this token to the list of non reserved keywords.

File: sql/plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -3602,6 +3603,80 @@ link_authentication
     : AUTHENTICATED BY user_object_name IDENTIFIED BY password_value
     ;
 
+// added by zrh
+create_database
+    : CREATE DATABASE database_name
+    ( USER SYS IDENTIFIED BY password_value
+     | USER SYSTEM IDENTIFIED BY password_value
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3610]: I suggest simplifying:

```suggestion
     ( USER (SYS | SYSTEM) IDENTIFIED BY password_value
```

File: sql/plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -3602,6 +3603,80 @@ link_authentication
...
+     | enable_pluggable_database
+    )+
+    ;
+
+database_logging_clauses
+    : LOGFILE database_logging_sub_clause (',' database_logging_sub_clause)*
+    | MAXLOGFILES UNSIGNED_INTEGER
+    | MAXLOGMEMBERS UNSIGNED_INTEGER
+    | MAXLOGHISTORY UNSIGNED_INTEGER
+    | (ARCHIVELOG | NOARCHIVELOG)
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3630]:
```suggestion
    | (MAXLOGFILES | MAXLOGMEMBERS | MAXLOGHISTORY) UNSIGNED_INTEGER
    | ARCHIVELOG
    | NOARCHIVELOG
```

File: sql/plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -3602,6 +3603,80 @@ link_authentication
...
+    ;
+
+database_logging_sub_clause
+    : (GROUP UNSIGNED_INTEGER)? file_specification
+    ;
+
+tablespace_clauses
+    : EXTENT MANAGEMENT LOCAL
+    | DATAFILE file_specification (',' file_specification)*
+    | SYSAUX DATAFILE file_specification (',' file_specification)*
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3641]:
```suggestion
    | SYSAUX? DATAFILE file_specification (',' file_specification)*
```

File: sql/plsql/PlSqlParser.g4 [hunk 5]
```diff
@@ -3602,6 +3603,80 @@ link_authentication
...
+// added by zrh
+create_database
+    : CREATE DATABASE database_name
+    ( USER SYS IDENTIFIED BY password_value
+     | USER SYSTEM IDENTIFIED BY password_value
+     | CONTROLFILE REUSE
+     | MAXDATAFILES UNSIGNED_INTEGER
+     | MAXINSTANCES UNSIGNED_INTEGER
+     | CHARACTER SET char_set_name
+     | NATIONAL CHARACTER SET char_set_name
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3615]:
```suggestion
     | (MAXDATAFILES | MAXINSTANCES) UNSIGNED_INTEGER
     | NATIONAL? CHARACTER SET char_set_name
```

File: sql/plsql/PlSqlParser.g4 [hunk 6]
```diff
@@ -3767,6 +3832,11 @@ dblink
     : database_name ('.' d=id_expression)* ('@' cq=id_expression)?
     ;
 
+// https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/DROP-DATABASE.html
+drop_database
+    : DROP DATABASE
+    ;
+
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3839]: The same parse rule already exists: https://github.com/antlr/grammars-v4/blob/ef698f1a02103d7e026ce7b3ea0b4120d4fc487c/sql/plsql/PlSqlParser.g4#L4060-L4063

Please reuse it.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

**Pull request closed without merging**